# **1. PATSTĀVĪGAIS DARBS. Ievads Machine Learning**


# 3. uzdevums - Lejupielādē datu kopu (dataset)
## Datu kopas iegūšana caur seaborn
## Datu kopa - TITANIKS

In [8]:
import seaborn as sns
import pandas as pd
df = sns.load_dataset('titanic')
print(f'Rindu skaits: {df.shape[0]}')
print(f'Kolonnu skaits: {df.shape[1]}')
df.head()

Rindu skaits: 891
Kolonnu skaits: 15


,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True


## Datu kopas iepazīšana

In [9]:
# Cik rindas un kolonnas?
df.shape

(891, 15)

In [10]:
# Kādas kolonnas ir?
df.columns.tolist()

['survived',
 'pclass',
 'sex',
 'age',
 'sibsp',
 'parch',
 'fare',
 'embarked',
 'class',
 'who',
 'adult_male',
 'deck',
 'embark_town',
 'alive',
 'alone']

In [11]:
# Datu tipi un trūkstošās vērtības
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   survived     891 non-null    int64   
 1   pclass       891 non-null    int64   
 2   sex          891 non-null    str     
 3   age          714 non-null    float64 
 4   sibsp        891 non-null    int64   
 5   parch        891 non-null    int64   
 6   fare         891 non-null    float64 
 7   embarked     889 non-null    str     
 8   class        891 non-null    category
 9   who          891 non-null    str     
 10  adult_male   891 non-null    bool    
 11  deck         203 non-null    category
 12  embark_town  889 non-null    str     
 13  alive        891 non-null    str     
 14  alone        891 non-null    bool    
dtypes: bool(2), category(2), float64(2), int64(4), str(5)
memory usage: 80.7 KB


In [12]:
# Pirmie statistikas rādītāji
df.describe()

,survived,pclass,age,sibsp,parch,fare
count,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


# 4. uzdevums - Definē ML problēmu

## **A. Problēmas apraksts**

**Problēma:** Prognozēt, vai pasažieris izdzīvos Titānika katastrofā, balstoties uz dzimumu, vecumu, biļetes klasi un citiem datiem.  

**Svarīgums:** Palīdz uzlabot drošības un evakuācijas procedūras, identificējot riska grupas.  

**Gala lietotājs:** Kuģa drošības komanda un analītiķi, kas veido evakuācijas plānus.  

**Mērķis:** Saprast, kuri faktori ietekmē izdzīvošanu un aprēķināt izdzīvošanas varbūtību.


## **B. ML uzdevuma tips**

**Uzdvuma veids:** Supervised learning  
- Tas ir supervised, jo mums ir zināmas **mērķa vērtības** (`survived`: 1 = izdzīvoja, 0 = neizdzīvoja). Modeļi tiek trenēti, lai prognozētu šo mērķa vērtību, izmantojot citus datus (features).

**Uzdevuma tips:** Klasifikācija  
- Tas ir klasifikācijas uzdevums, jo mērķis ir prognozēt **kategorisku iznākumu** — vai pasažieris izdzīvos (1) vai neizdzīvos (0).


## **C. Target variable**

Šajā sadaļā definēsim **target mainīgo**, tas ir, vērtību, ko mēģinām prognozēt ar mūsu ML modeli.  
Šajā uzdevumā target mainīgais ir `survived`, kas norāda, vai pasažieris izdzīvoja (1) vai neizdzīvoja (0).

In [13]:
# Target mainīgais
print(df['survived'].value_counts())  # cik pasažieru izdzīvoja/neizdzīvoja
print(f'\nIzdzīvoja: {df["survived"].mean():.1%}')  # procentuāli

survived
0    549
1    342
Name: count, dtype: int64

Izdzīvoja: 38.4%


**Paskaidrojums:**  

**Target mainīgais (`survived`)** ir mūsu mērķis, ko mēģinām prognozēt datu analīzē vai mašīnmācīšanās uzdevumā.

**Kolonna `survived`:**  
- Kolonna `survived` ir target mainīgais, ko mēģinām prognozēt.  
- Vērtības:  
  - `0` – pasažieris neizdzīvoja  
  - `1` – pasažieris izdzīvoja

**Lietojums Python kodā:**  
- `df['survived']` - kolonna “survived” datu rāmī `df`.  
- `df['survived'].value_counts()` - parāda, cik cilvēki izdzīvoja (1) un cik neizdzīvoja (0).  
- `df["survived"].mean()` - aprēķina procentuālo daļu izdzīvojušo:  
  - Ja vērtības ir 0 un 1, vidējā vērtība = izdzīvojušo īpatsvars.  
  - `:.1%` → formatē skaitli procentos ar vienu zīmi aiz komata.

## **D. Features saraksts**

## Features (iezīmes) datu modelēšanai

### Skaitliskās (numerical)
- age - cilvēka vecums
- fare - biļetes cena
- sibsp - brāļu/māsu un partneru skaits
- parch - vecāku/bērnu skaits

### Kategoriskās (categorical)
- sex - dzimums
- pclass - kuģa klase
- embarked - iekāpšanas osta
- alone - vai cilvēks ceļoja viens



In [14]:
# Features pārskats
print('Skaitliskās features:')
print(df.select_dtypes(include='number').columns.tolist())
# select_dtypes(include='number') - atlasīs visas numeriskās kolonnas
print('\nKategoriskās features:')
print(df.select_dtypes(include='category').columns.tolist())
# select_dtypes(include='category') - atlasīs visas kategoriskās kolonnas
# columns.tolist() - izdrukā kolonnas kā sarakstu

Skaitliskās features:
['survived', 'pclass', 'age', 'sibsp', 'parch', 'fare']

Kategoriskās features:
['class', 'deck']


**Paskaidrojums:**  

- **Skaitliskās features** – mainīgie ar skaitliskām vērtībām, kur var veikt aprēķinus un statistiku.  
- **Kategoriskās features** – mainīgie, kas satur kategorijas vai grupas, piemēram, dzimums vai biļetes klase.  
- ML modelī šīs features tiek izmantotas, lai prognozētu target mainīgo (`survived`).

## **E. Gala secinājumi / piezīmes**

**1.** Datu bāzē **titanic** ir informācija par 891 pasažieri, kurus var raksturot ar 15 pazīmēm.

**2.** Šajā nedēļā mēs definējām **target mainīgo (`survived`)** un izpētījām tā vērtību sadalījumu:
  - Izdzīvoja (`1`): aptuveni 38% pasažieru.
  - Neizdzīvoja (`0`): aptuveni 62% pasažieru.

**3.** Identificējām **features**, sadalot tās skaitliskajās un kategoriskajās:
  - **Skaitliskās:** `age`, `fare`, `sibsp`, `parch`
  - **Kategoriskās:** `sex`, `pclass`, `embarked`, `alone`

**4.** Šīs zināšanas palīdz sagatavot datus ML modelim un izvēlēties piemērotus klasifikācijas algoritmus, lai prognozētu izdzīvošanu jaunajiem pasažieriem.
